pkg/pillar/zedcloud/lookupproxy.go

```diff
@@ -121,3 +121,51 @@ func LookupProxy(status *types.DeviceNetworkStatus, ifname string,
 	log.Infof("LookupProxy: No proxy configured for port %s", ifname)
 	return nil, nil
 }
+
+// IntfLookupProxyCfg - check if the intf has proxy configured
+func IntfLookupProxyCfg(status *types.DeviceNetworkStatus, ifname, downloadURL string) string {
+	var mayChange bool
+	// if proxy is not on the intf, then don't change anything
+	// if download URL has "http://" or "https://" then no change here regardless of proxy
+	// if there is proxy on this intf, treat empty url scheme as for https,
```

Comment:
That's not what the code does; it first looks for a proxy as if it was https://, and if not found it looks for a proxy as if it was http://

Reply:
Right, i should have said prefer https://